tests/unit/test_charm.py

```diff
@@ -78,12 +108,63 @@ def test_layer_updated(harness) -> None:
                 "command": "identity_platform_login_ui",
                 "startup": "enabled",
                 "environment": {
-                    "HYDRA_ADMIN_URL": TEST_HYDRA_URL,
-                    "KRATOS_PUBLIC_URL": TEST_KRATOS_URL,
+                    "HYDRA_ADMIN_URL": "http://hydra-public-url:80/testing-hydra",
+                    "KRATOS_PUBLIC_URL": "http://kratos-public-url:80/testing-kratos",
```

Comment:
Don't hard code this, get the data from the relation and match the pebble layer against those.